cpp/README.md

```diff
@@ -311,6 +311,45 @@ ctest -L gandiva
 This library is still in Alpha stages, and subject to API changes without
 deprecation warnings.
 
+### Building and developing Flight (optional)
+
+In addition to the Arrow dependencies, Flight requires:
+* gRPC
+* Protobuf
+* c-ares
+
+By default, Arrow will try to download and build these dependencies
+when building Flight.
+
+The optional `flight` libraries and tests can be built by passing
+`-DARROW_FLIGHT=ON`.
+
+```shell
+cmake .. -DARROW_FLIGHT=ON -DARROW_BUILD_TESTS=ON
+make
+```
+
+You can also use existing installations of the extra dependencies.
+When building, set the environment variables `GRPC_HOME` and/or
+`PROTOBUF_HOME` and/or `CARES_HOME`.
+
+On Ubuntu/Debian, you can install these dependencies with:
```

Comment:
Something missing here? On Ubuntu 18.04, one might get away with (untested):
```
apt install libgrpc++-dev
```

Reply:
Looks like I'm being scatterbrained today; I meant to spin up a VM to test this before I added it to the readme.

Reply:
Looks like you need several packages, and Debian/Ubuntu statically links one of the gRPC libraries into the others, which adds another permutation... the README/FindgRPC now cover this.

Reply:
aaaand 18.04 ships too old a version of gRPC, let me follow up on this...

Reply:
I'd be a bit shocked if we were able to use Linux system packages. gRPC has been too fast-moving

Reply:
Fair enough. Maybe I can mention these variables, but say it's at the user's own risk?